pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -603,6 +602,26 @@ private void updateBundleData() {
                 LoadManagerShared.fillNamespaceToBundlesMap(preallocatedBundleData.keySet(), namespaceToBundleRange);
             }
         }
+
+        // Remove not active bundle from loadData
+        for (String bundle : bundleData.keySet()) {
+            if (!activeBundles.contains(bundle)) {
+                long notActiveTimes = notActiveBundleCounter
+                        .computeIfAbsent(bundle, k -> new AtomicLong()).incrementAndGet();
+
+                if (notActiveTimes > nonActiveBundleDeleteThreshold) {
```

Comment:
Do we know what causes inconsistency between `activeBundles` and `bundleData.keySet()`?

Reply:
Do we need to worry about other data inconsistencies, for example, between this bundleData(and deleteBundleDataFromMetadataStore) vs other data structures?

Reply:
> Do we know what causes inconsistency between `activeBundles` and `bundleData.keySet()`?

Yes, the bundleData is upload to metadataStore by `LoadReportUpdaterTask` which will upload broker load data and bundle data. 

This task will be execute every 5 seconds by default and also need meets the `ModularLoadManagerImpl.needBrokerDataUpdate` condition.

And the broker local bundle data is only update by `BrokerService.updateRates()` every 60 seconds.  

So if one broker load the bundle but not update the bundleStats in time, 
and the previous bundle owner update the bundleStats and triggered the `LoadReportUpdaterTask` to report its own load data. 

The bundle data is missing in all broker load-data.

Reply:
> 

In my opinion. the undelete bundle data won't harm the other logic. most logic is just to query the bundle data. If bundle is splited. the system won't to do ops for this bundle again.

Reply:
> Do we need to worry about other data inconsistencies, for example, between this bundleData(and deleteBundleDataFromMetadataStore) vs other data structures?

For the time now. I think the bundleData is eventual consistency to create a cluster wide bundle load view.
which means that the bundleData can be late but should be on the metadataStore.

The undelete bundle data won't harm the other logic. most logic is just to query the bundle data. If bundle is splited. the system won't to do other operation for this bundle again.